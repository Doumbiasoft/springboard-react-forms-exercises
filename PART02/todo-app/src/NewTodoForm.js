import React, { useState, useEffect } from 'react';

const NewTodoForm = ({ addTodo, editingTask, editTodo }) => {
  const [task, setTask] = useState('');

  useEffect(() => {
    if (editingTask) {
      setTask(editingTask);
    }
  }, [editingTask]);

  const handleSubmit = (e) => {
    e.preventDefault();
    if (editingTask) {
      editTodo(task);
    } else {
      addTodo({ task });
    }
    setTask('');
  };

  return (
    <form onSubmit={handleSubmit}>
      <label>
        New Task:
        <input
          type="text"
          value={task}
          onChange={(e) => setTask(e.target.value)}
        />
      </label>
      <button>{editingTask ? 'Edit Todo' : 'Add Todo'}</button>
    </form>
  );
};

export default NewTodoForm;
