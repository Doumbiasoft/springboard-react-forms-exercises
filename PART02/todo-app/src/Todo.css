.Todo{
    margin: 20px;
}
.Todo-btn{
    margin-left: 10px;
}