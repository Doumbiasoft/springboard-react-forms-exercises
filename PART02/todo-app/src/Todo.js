import React from 'react';
import './Todo.css'

const Todo = ({ id, task, onEdit, onRemove }) => {
  return (
    <div className="Todo">
      {task}
      <button className='Todo-btn' onClick={() => onEdit(task)}>Edit</button>
      <button className='Todo-btn' onClick={() => onRemove(id)}>X</button>
    </div>
  );
};

export default Todo;
