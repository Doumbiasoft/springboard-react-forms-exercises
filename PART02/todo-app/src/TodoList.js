import React, { useState } from 'react';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';
import { v4 as uuid } from 'uuid';

const TodoList = () => {
  const [todos, setTodos] = useState([]);
  const [editingTask, setEditingTask] = useState(null);

  const addTodo = (todoData) => {
    setTodos([...todos, {...todoData, id:uuid()}]);
  };

const editTodo = (newTask) => {
    const editedTodos = todos.map((todo) =>
      todo.task === editingTask ? { ...todo, task: newTask } : todo
    );
    setTodos(editedTodos);
    setEditingTask(null);
  };

  const removeTodo = (id) => {
    const newTodos = [...todos];
    setTodos(newTodos.filter(toto => toto.id !== id));
  };

  return (
    <div>
      <NewTodoForm
       addTodo={addTodo}
       editTodo={editTodo}
       editingTask={editingTask}
      />
      {todos.map((todo) => (
        <Todo
          id={todo.id}
          key={todo.id}
          task={todo.task}
          onEdit={setEditingTask}
          onRemove={removeTodo}
        />
      ))}
    </div>
  );
};

export default TodoList;
