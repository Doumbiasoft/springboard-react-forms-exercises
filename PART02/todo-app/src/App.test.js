import { render,fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';
import TodoList from './TodoList';

describe("Smoke Test", ()=>{

  it('should not crash when rendering', () => {
    render(<NewTodoForm />);
  
  });
  it('should not crash when rendering', () => {
    render(<Todo />);
  });
  it('should not crash when rendering', () => {
    render(<TodoList />);
  });
});

describe("Snapshot Test", ()=>{
  it("should matches snapshot", ()=> {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should matches snapshot", ()=> {
    const { asFragment } = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should matches snapshot", ()=> {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
  });
  
});
