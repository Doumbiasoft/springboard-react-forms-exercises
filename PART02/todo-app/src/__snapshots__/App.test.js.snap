// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test should matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label>
      New Task:
      <input
        type="text"
        value=""
      />
    </label>
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;

exports[`Snapshot Test should matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <button
      class="Todo-btn"
    >
      Edit
    </button>
    <button
      class="Todo-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;

exports[`Snapshot Test should matches snapshot 3`] = `
<DocumentFragment>
  <div>
    <form>
      <label>
        New Task:
        <input
          type="text"
          value=""
        />
      </label>
      <button>
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
